src/api/java/com/minecolonies/api/crafting/ImmutableItemStorage.java

```diff
@@ -0,0 +1,27 @@
+package com.minecolonies.api.crafting;
+
+import net.minecraft.item.ItemStack;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Immutable ItemStorage version.
+ */
+public class ImmutableItemStorage extends ItemStorage
```

Comment:
its nice that the setamount can now be immutable, but wasnt the most common problem that the internal stack got modified? which we still can do with this version. should we maybe make an ItemStack wrapper for that case?

Reply:
The actual problems I fixed were that the count of this storage were changed. I did check into ItemStack wrappers, but then thought that would become an overkill. It's because on creating a new ItemStack we gotta copy the old one, which needs to go in the constructor of the itemStack which itself calls the capability events which can be very slow if mods mess it up.

Reply:
I know we'd talked about possibly doing some corruption checks on the retrieval of the item stack from the ItemStorage. It should be as simple as checking that if the amount is less than maxstack, the count == amount, etc.. 

At least then we'd have early warning that the internal stack was getting modified when it shouldn't be.

Reply:
I was thinking about easy ways to detecting it, but couldn't find any easy way that couldn't break in valid cases.